File: algorithms/typescript/BT-postorder.ts
function post_walk(curr: BinaryNode<number> | null, path: number[]): number[] {
  if (!curr) {
    return path;
  }
  post_walk(curr.left, path);
  post_walk(curr.right, path);
  path.push(curr.value);
  return path;
}
function post_order_search(head: BinaryNode<number>, path: number[]): number[] {
  return post_walk(head, []);
}

File: algorithms/python/bt-postorder.py
class BinaryNode:
    def __init__(self) -> None:
        self.value: int
        self.left: BinaryNode
        self.right: BinaryNode


def walk(curr: BinaryNode | None, path: list[int]) -> list[int]:
    if not curr:
        return path
    walk(curr.left, path)
    walk(curr.right, path)
    path.append(curr.value)
    return path


def post_order_search(head: BinaryNode) -> list[int]:
    return walk(head, [])
